File: Back-end/ai-server/src/main/java/com/lht/pojo/Food.java
package com.lht.pojo;

import com.pgvector.PGvector;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.Type;

import java.util.UUID;

@Entity
@Table(name = "foods")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Food {

    @Id
    private UUID uuid;

    private int code;

    private String name;

    private String category;

    @Column(name = "calories_100g")
    private Float calories100g;

    @Column(name = "embedding", columnDefinition = "vector(768)")
    @Type(PGvectorType.class)
    private PGvector embedding;
}

File: Front-end/app/lib/models/Food.dart
class Food {
  final String? uuid;
  final int? code;
  final String? name;
  final String? category;
  final double? calories100g;

  Food({
    this.uuid,
    this.code,
    this.name,
    this.category,
    this.calories100g,
  });

  factory Food.fromJson(Map<String, dynamic> json) => Food(
    uuid: json['uuid'],
    code: json['code'],
    name: json['name'],
    category: json['category'],
    calories100g: (json['calories100g'] as num?)?.toDouble(),
  );
}